packages/fiori-elements-writer/src/types.ts

```diff
@@ -0,0 +1,96 @@
+import { Ui5App, App } from '@sap-ux/ui5-application-writer';
+import { OdataService } from '@sap-ux/odata-service-writer';
+
+export enum TemplateType {
+    Worklist = 'worklist',
+    ListReportObjectPage = 'lrop',
+    AnalyticalListPage = 'alp',
+    OverviewPage = 'ovp',
+    FormEntryObjectPage = 'feop'
+}
+
+/**
+ * General validation error thrown if app config options contain invalid combinations
+ */
+export class ValidationError extends Error {
+    /** 
+     * ValidationError constructor.
+     * 
+     * @param message - the error message
+     */
+    constructor(message: string) {
+        super(`Validation error: ${message}`);
```

Comment:
Maybe i18n here?

Reply:
Not sure since it's the name of an error rather than the message. The error message is translated/localised.